pkg/mcs/tso/server/server.go

```diff
@@ -99,16 +128,175 @@ func (s *Server) Context() context.Context {
 
 // Run runs the TSO server.
 func (s *Server) Run() error {
+	go systimemon.StartMonitor(s.ctx, time.Now, func() {
+		log.Error("system time jumps backward", errs.ZapError(errs.ErrIncorrectSystemTime))
+		timeJumpBackCounter.Inc()
+	})
+	if err := s.initClient(); err != nil {
+		return err
+	}
+
+	s.serverLoopCtx, s.serverLoopCancel = context.WithCancel(s.ctx)
+	if err := s.startServer(s.ctx); err != nil {
+		return err
+	}
+
+	s.startServerLoop(s.ctx)
+
 	return nil
 }
 
+func (s *Server) startServerLoop(context.Context) {
+	s.serverLoopWg.Add(2)
+	go s.primaryElectionLoop()
+	go s.tsoAllocatorLoop()
+}
+
+// tsoAllocatorLoop is used to run the TSO Allocator updating daemon.
+func (s *Server) tsoAllocatorLoop() {
+	defer logutil.LogPanic()
+	defer s.serverLoopWg.Done()
+
+	ctx, cancel := context.WithCancel(s.serverLoopCtx)
+	defer cancel()
+	s.tsoAllocatorManager.AllocatorDaemon(ctx)
+	log.Info("tso server is closed, exit allocator loop")
+}
+
+func (s *Server) primaryElectionLoop() {
```

Comment:
Actually, I think we need the tenant-level election loop, instead of the server-level.

Reply:
For the first milestone, we implement server-level loop as the special case of keyspace-group-level with only 1 keyspace group.

Reply:
We'll still use mcs server/service discovery path.